src/components/input/use-input-value.tsx

```diff
@@ -38,7 +51,21 @@ export const useInputValue: UseInputValue = ({ value = "", onChange, maxChars })
     [isDirty, maxChars, onChange]
   )
 
-  const maxCharsIndicator = maxChars ? `${inputValue.length}/${maxChars}` : ""
+  const maxCharsIndicator = useMemo(() => (maxChars ? `${inputValue.length}/${maxChars}` : ""), [
+    maxChars,
+    inputValue,
+  ])
```

Comment:
@Azein not sure if that simple operation needs `useMemo` (maybe `useMemo` perf cost is bigger)

Reply:
Almost no single operation needs useMemo to not do the operation, unless it's some very big iteration. The purpose is not to trigger re-render of the tree that will use the maxCharsIndicator value from this hook, which may have considerable cost (or may not, but this is the UI Kit level, so we probably should care).

We are still not doing this in all cases, but we should start to care about memo-ing the stuff we expose, I think.

Reply:
i don't understand, when you return primitives, `useMemo` shouldn't impact rerender, or am i missing something?